### DISCUSSION TITLE
`UnetBasicBlock` does not use bias for convolutions

### DISCUSSION THREAD

**Discussion by function2-llx** [Project-MONAI/MONAI#5644] [General]
I find that the implementation of `UnetBasicBlock` does not use bias for its convolutions. E.g., it uses the default value for `bias` (by not specifying it) :

https://github.com/Project-MONAI/MONAI/blob/00a64346c57e62accc36a04a41ae20303f1ecf3e/monai/networks/blocks/dynunet_block.py#L142-L152

while the default value of `bias` is `False` according to the definition of the `get_conv_layer` function.

https://github.com/Project-MONAI/MONAI/blob/00a64346c57e62accc36a04a41ae20303f1ecf3e/monai/networks/blocks/dynunet_block.py#L267-L277

However, the nnU-Net (the model that `DynUNet` is based on) uses bias by default: https://github.com/MIC-DKFZ/nnUNet/blob/aa53b3b87130ad78f0a28e6169a83215d708d659/nnunet/network_architecture/generic_UNet.py#L112-L113

I'm also not able to find anything mentioning this behavior in the paper [_Optimized U-Net for Brain Tumor Segmentation_](https://arxiv.org/pdf/2110.03352.pdf) (by searching text `bias`).

On the other hand, however, the implementation of most of the convolution-based models (e.g., ResNet, UperNet) do not use bias.

Could anyone please give any explanation on this? Thanks!

**Comment by function2-llx** [comment]
Find some plausible explanation [from the Internet](https://stackoverflow.com/a/65295983/9815239): there's usually a normalization layer after the convolution, which has a bias term.

**Reply by wyli** [reply]
yes, and the learnable parameters may or may not be enabled by default (for example in batchnorm vs instancenorm), please see https://github.com/Project-MONAI/MONAI/issues/5222 https://github.com/pytorch/pytorch/issues/22755

**Reply by function2-llx** [reply]
Thanks for the additional information. I never knew of such an issue with InstanceNorm, though I've used it for a long time.
